component-catalog-connectors/build-a-custom-connector.md

```diff
@@ -1,3 +1,215 @@
+
+Irrespective of which type of connector you plan to implement, you need to complete the following tasks:
+ - create a catalog connector schema
+ - create a schema provider
+ - implement the catalog connector, which queries the catalog and retrieves components from the catalog
+ - package the catalog connector
+
+To get you started quickly with catalog connector development, the Elyra examples repository includes [quickstart catalog connector template files](connector-template/), which are referenced in the instructions below.
+
+The quickstart connector contains a built-in catalog, which contains only a single component. The aforementioned Kubeflow Pipelines and Apache Airflow component examples connectors are more flexible versions of this connector, which can serve many components.
+
+### Prepare for development
+
+1. Install Elyra version 3.3 or later from PyPI or source.
+1. Clone or fork the [Elyra examples repository](https://github.com/elyra-ai/examples).
+1. Change into the `component-catalog-connectors/connector-template` directory. The directory contains a fully functional connector implementation that can be customized to meet your needs.
+1. Verify that the quickstart template connector works as expected by completing the [installation and usage instructions in the README](/component-catalog-connectors/connector-template/README.md).
+
+### Create a catalog connector schema
+
+All catalog connectors have in common that they use a schema to define their internal and public properties. The public properties are exposed to the user in the catalog connector configuration, as shown for the quickstart template connector below:
+
+![Quickstart catalog connector configuration](doc/images/quickstart-catalog-configuration.png)
+
+The common public properties are:
+ - "learn more" link, providing access to to connector documentation
+ - catalog instance name
+ - catalog instance description
+ - runtime type filter, which governs the kind of components the connector makes available
+ - component categories, which group the components together in the pipeline editor
+
+Connector schemas can optionally also include custom public properties. For illustrative purposes, the quickstart template connector defines a required custom property and an optional custom property. Required properties are typically used to collect catalog connectivity information, such as a server URL, or API token. 
+
+#### Customize the schema
+
+The quickstart connector default schema file `TODO-catalog.json` is located in the [`todo_catalog_connector`](/component-catalog-connectors/connector-template/todo_catalog_connector/) directory.
+1. Rename the file to reflect your catalog type.
+1. Using an IDE of your choice, find and replace all occurrences of '`TODO-catalog.json`' in all files in the `component-catalog-connectors/connector-template` directory. (There should be seven.)
```

Comment:
I think there are only 3 occurrences of `TODO-catalog.json` in `component-catalog-connectors/connector-template`

Reply:
let me change that to several (: